src/Orleans/Storage/OrleansRelationalExtensions.cs

```diff
@@ -0,0 +1,1506 @@
+                command.Parameters.Add(suspectingSilosParameter);
+
+                var suspectingTimesParameter = CreateSuspectingTimesParameter(command, membershipEntry, direction);
+                command.Parameters.Add(suspectingTimesParameter);
+
+            }, (selector, _) => { return selector.GetBoolean(0); }).ConfigureAwait(continueOnCapturedContext: false);
+
+            return ret.First();
+        }
+
+
+        private static ReminderEntry CreateReminderEntry(IDataRecord record)
+        {
+            //Having non-null field, GrainId, means with the query filter options, an entry was found.
+            if(record.GetValueOrDefault<string>("GrainId") != null)
+            {
+                return new ReminderEntry
+                {
+                    GrainRef = GrainReference.FromKeyString(record.GetValue<string>("GrainId")),
+                    ReminderName = record.GetValue<string>("ReminderName"),
+                    StartAt = record.GetValue<DateTime>("StartTime"),
+                    Period = TimeSpan.FromMilliseconds(record.GetValue<int>("Period")),
+                    ETag = Convert.ToBase64String(record.GetValue<byte[]>("ETag"))
+                };
+            }
+
+            return null;
+        }
+
+
+        private static Tuple<MembershipEntry, string, int, string> CreateMembershipEntry(IDataRecord record)
+        {
+            //Retrieving by ordinal is slightly (immeasurably?) faster. Perhaps something to refactor.                        
+
+            //TODO: This is a bit of hack way to check in the current version if there's membership data or not, but if there's a start time, there's member.            
+            DateTime? startTime = record.GetValueOrDefault<DateTime?>("StartTime");
+
+            int tableVersion = (int)record.GetValueOrDefault<long>("Version");
+            string versionETag = Convert.ToBase64String(record.GetValueOrDefault<byte[]>("VersionETag"));
+            string etag = null;
+            MembershipEntry entry = null;
+            if(startTime.HasValue)
+            {
+                int port = record.GetValue<int>("Port");
+                int generation = record.GetValue<int>("Generation");
+                string address = record.GetValue<string>("Address");
+                string hostName = record.GetValueOrDefault<string>("HostName");
+                SiloStatus status = record.GetValue<SiloStatus>("Status");
+                int? proxyPort = record.GetValueOrDefault<int?>("ProxyPort");
+                bool? isPrimary = record.GetValueOrDefault<bool?>("Primary");
+                string roleName = record.GetValue<string>("RoleName");
+                string instanceName = record.GetValue<string>("InstanceName");
+                int? updateZone = record.GetValue<int?>("UpdateZone");
+                int? faultZone = record.GetValueOrDefault<int?>("FaultZone");
```

Comment:
ReSharper flags this as unused variable, but is really missing assignment into MembershipEntry.FaultZone at line 832 below.